### DISCUSSION TITLE
Suivi des dépenses quotidiennes

### DISCUSSION THREAD

**Discussion by Adamou02** [MadeInPierre/finalynx#142] [Q&A] [answered]
Petite demande simple et rapide, on m'a parlé d'un "Suivi des dépenses quotidiennes" dans cette [discussion](https://community.finary.com/t/fonctionnement-de-powens-budget-insight/8657/5) Finary.

Quel est le rendu visuel de ce suivi de dépenses (je n'arrive pas à faire fonctionner/trouver un Suivi des dépenses quotidiennes avec finalynx) ? 

Merci <3

**Comment by MadeInPierre** [comment, marked as answer]
Salut !

Alors déjà pour contexte, j'avais démarré Finalynx pour automatiser mes décisions de "quoi investir où", et j'ai ensuite étendu progressivement l'idée en fonction de mes envies / demandes d'autres gens ^^ simulation future, dépenses quotidiennes, ...

Récemment j'ai trouvé une API semi-officielle pour N26 et ça m'a donné envie d'organiser mes dépenses quotidiennes. J'avais 2 problèmes perso à résoudre : 
- Ça m'arrive souvent de payer un resto (et autres) pour 2 et l'autre personne me rembourse plus tard, donc certaines transactions ont un montant qui ne correspond pas à ce que j'ai payé pour moi-même.
- Je voulais surtout classifier mes dépenses par importance : fixe, variable, pour le fun, ...

J'ai donc fait une petite extension (qui pour l'instant est séparée du reste de Finalynx, tout est dans le dossier `finalynx/budget` et est raccordé à la console dans `finalynx/assistant.py`), codée rapidement donc qui en est à ses débuts. L'idée est de stocker les transactions de N26 (ou ailleurs) dans un Google Sheets perso qui sert de base de données, et on peut classifier la dépense depuis la console de Finalynx interactivement (ou directement sur le Sheets). Ensuite Finalynx affiche quelques stats sur la console :

![IMG_20230901_115127_838.png](https://github.com/MadeInPierre/finalynx/assets/12670280/5ed59924-a582-466c-8ef7-df5e315aa9ae)



2023 correspond aux dépenses notées comme annuelles (cadeaux, vêtements, ...), rapporté à un montant mensuel en bleu. Ensuite les dépenses mensuelles sont additionnées pour chaque mois, et on rajoute les dépenses annuelles en bleu. Enfin, on a notre moyenne de dépenses en bas et à quelle distance de la moyenne on est pour le mois en cours.

Tout ceci est complètement modifiable, n'hésite pas à proposer des changements pour tes besoins perso ! Bien sûr ce serait aussi cool de voir un nouveau projet sur GitHub avec ton approche si tu ne cherches pas la même chose ^^

**Reply by MadeInPierre** [reply]
D'ailleurs il y a un dashboard web très moche pour l'instant, si ça te tente de le modifier/remplacer par quelque chose de mieux 🙃 notamment toutes sortes de graphiques et stats sur les dépenses quotidiennes !

**Reply by MadeInPierre** [reply]
Dernière chose : il y a un peu de setup à faire pour le budget quotidien (principalement créer son Google Sheets et avoir un token API), si ça t'intéresse je peux rédiger un tuto de setup :) je crois que je suis le seul à me servir de la partie budget pour l'instant.

**Reply by Adamou02** [reply]
Je ne suis pas contre un petit tuto, surtout pour l'utilisation du token API. Je n'ai pas eu le temps d'utiliser finalynx pour le moment, et comme je maitrise mal les API, ca pourrait m'être utile pour ce sujet mais aussi pour ton programme.

J'avais commencer une template Excel pour ma gestion de budget. Je vais voir si je peut adapter ma solution pour l'intégrer sur finalynx/budget

L'excel en question : [Gestion Budget_dev.xlsx](https://github.com/MadeInPierre/finalynx/files/12497757/Gestion.Budget_dev.xlsx)

**Reply by MadeInPierre** [reply]
Ca marche je rédige ça dès que possible ! Sorry pour mon délai de réponse, vacances en ce moment ^^ 

Sympa ton template, j'ai l'impression qu'il est assez facile à intégrer à ce que j'ai déjà fait ici. Tu souhaites surtout organiser tes dépenses selon des catégories c'est ça (e.g. emprunt, courses, animaux, ...) ? Si oui, il suffirait d'ajouter un champ "Catégorie" dans la liste des infos qu'on rentre pour décrire chaque dépense et le tour est joué. Bien sûr, à condition d'ajouter La Banque Postale comme source si tu n'utilises pas N26.

P.S. Si tu dépenses vraiment 11k€/mois, je crois qu'il y a d'autres logiciels payants bien plus performants que ce petit finalynx :smile:

**Comment by Adamou02** [comment]
Je marquerai comme résolu (après ton tuto setup), je n'ai même pas pris le temps de lire en profondeur ton dossier [docs](https://github.com/MadeInPierre/finalynx/tree/main/docs)

Je rouvrirai la discussion si je ne trouve pas mes réponses là bas.
Merci en tt cas !

**Reply by MadeInPierre** [reply]
Avec plaisir ^^ Je te tiens au courant quand c'est publié, en attendant le dossier docs est dispo en version jolie sur le [site de documentation](https://finalynx.readthedocs.io/en/latest/)

**Reply by MadeInPierre** [reply]
Hey! Me voici de retour ^^ Si tu es toujours intéressé, j'ai mis à jour les tutoriels avec un nouveau dédié aux dépenses quotidiennes :

https://finalynx.readthedocs.io/en/latest/quickstart/tutorials.html

J'espère que ce sera clair, n'hésite pas à me dire s'il manque quelque chose !
